### PR TITLE
Document OpenTelemetry @WithSpan

### LINKED ISSUE #23307: Tracing additional methods in OpenTelemetry
### Description

Possibility to trace additional methods other than REST calls.
Lack of this functionality prohibits from migrating to open telemetry from open tracing(which is deprecated).

### Implementation ideas

Functionality similar to 
https://quarkus.io/guides/opentracing#tracing-additional-methods

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes https://github.com/quarkusio/quarkus/issues/23307.

### CODE REVIEW

File: docs/src/main/asciidoc/opentelemetry.adoc [hunk 1]
```diff
@@ -337,6 +337,41 @@ These sections will outline what is necessary to propagate traces across process
 
 The instrumentation documented in this section has been tested with Quarkus and works in both standard and native mode.
 
+=== CDI
+
+Annotating a method in any CDI aware bean with the `io.opentelemetry.extension.annotations.WithSpan`
+annotation. This will create a new Span and establish any required relationships with the current Trace context.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/opentelemetry.adoc:343]:
```suggestion
annotation will create a new Span and establish any required relationships with the current Trace context.
```

**Review by gastaldi — CHANGES_REQUESTED** [review]
It would be nice if the guide explained (or had some links pointing to) what a Span and a Trace are. As a newbie, I have no idea what that is :)

**Comment by radcortez** [comment]
> It would be nice if the guide explained (or had some links pointing to) what a Span and a Trace are. As a newbie, I have no idea what that is :)

Hum, for that I think we need to rewrite the guide. Let me do that on a separate PR.

**Review by gastaldi — APPROVED** [review]

**Comment by geoand** [comment]
Best squash the commits before merging

**Pull request merged**